build-logic/documentation/src/main/groovy/gradlebuild/docs/GradleJavadocsPlugin.java

```diff
@@ -145,4 +149,62 @@ private void generateJavadocs(Project project, ProjectLayout layout, TaskContain
             task.getReports().getXml().getOutputLocation().set(new File(checkstyle.getReportsDir(), "checkstyle-api.xml"));
         });
     }
+
+    /**
+     * Used to bridge Gradle 8 and Gradle 9 APIs for Gradleception.
+     *
+     * TODO: Remove after Gradle 9
+     */
+    private static class JavadocSupport {
+
+        private final Javadoc javadoc;
+
+        public JavadocSupport(Javadoc javadoc) {
+            this.javadoc = javadoc;
+        }
+
+        public void setTitle(String title) {
+            setProperty("setTitle", title);
+        }
+
+        public void setClasspath(FileCollection classpath) {
+            setProperty("setClasspath", classpath);
+        }
+
+        public void setDestinationDir(File destinationDir) {
+            setProperty("setDestinationDir", destinationDir);
+        }
+
+        private <T> void setProperty(String setterName, T value) {
+            if (GradleVersion.current().compareTo(GradleVersion.version("9.0")) < 0) {
+                setPropertyForGradleLth9(setterName, value);
+            } else {
+                setPropertyForGradleGth9(setterName, value);
+            }
+        }
+
+        private <T> void setPropertyForGradleGth9(String setterName, T value) {
+            try {
+                // Javadoc is Javadoc_Decorated, so it has set<Property>(Object) setter
+                javadoc.getClass().getMethod(setterName, Object.class).invoke(javadoc, value);
+            } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        private <T> void setPropertyForGradleLth9(String setterName, T value) {
```

Comment:
❓ Does `<T>` add anything here instead of just passing an `Object`?

Reply:
No, not really, I can change it to Object

Reply:
Done